src/oneD/StFlow.cpp

```diff
@@ -513,18 +545,49 @@ void StFlow::updateDiffFluxes(const doublereal* x, size_t j0, size_t j1)
             double wtm = m_wtm[j];
             double rho = density(j);
             double dz = z(j+1) - z(j);
-            for (double k = 0; k < m_nsp; k++) {
-                m_flux(k,j) = m_wt[k]*(rho*m_diff[k+m_nsp*j]/wtm);
-                m_flux(k,j) *= (X(x,k,j) - X(x,k,j+1))/dz;
-                sum -= m_flux(k,j);
+            for (size_t k = 0; k < m_nsp; k++) {
+                if ( k != kE ){
+                    m_flux(k,j) = m_wt[k]*(rho*m_diff[k+m_nsp*j]/wtm);
+                    m_flux(k,j) *= (X(x,k,j) - X(x,k,j+1))/dz;
+                    sum -= m_flux(k,j);
+                } else {
+                    //The approximation of the diffusion coeff. of electron
+                    //equation (28) with Einstein's relation
+                    //D_e = (m_H3Ox / m_E ) ^ 0.5 * D_H3Ox
+                    //reference:
+                    //Han, Jie, et al. "Numerical modelling of ion transport in flames.
+                    //" Combustion Theory and Modelling 19.6 (2015): 744-772.
+                    m_flux(k,j) = m_wt[k]*(rho*pow(m_wt[kH3Ox]/m_wt[kE],0.5)
```

Comment:
- Doesn't this expression still need to be multiplied by the gradient in electron mole fraction, and update `sum` in the same way as the normal case?
- Is there a way to do this by setting the values of the transport coefficient for the electron, rather than introducing a special case here?
- This needs to handle the case where `E` is a species but `H3O+` is not without crashing.